tests/test_signing.py

```diff
@@ -0,0 +1,90 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+
+import tempfile
+from collections.abc import Mapping
+from pathlib import Path
+
+import pytest
+
+from mkosi.run import find_binary, run
+
+from . import Image, ImageConfig
+
+pytestmark = pytest.mark.integration
+
+
+def test_signing_checksums_with_sop(config: ImageConfig) -> None:
+    if find_binary("sqop", root=config.tools) is None:
+        pytest.skip("Needs `sqop` binary in PATH to perform sop tests.")
+    with tempfile.TemporaryDirectory() as path, Image(config) as image:
+        tmp_path = Path(path)
+        tmp_path.chmod(0o755)
+
+        signing_key = tmp_path / "signing-key.pgp"
+        signing_cert = tmp_path / "signing-cert.pgp"
+
+        # create a brand new signing key
+        with open(signing_key, "wb") as o:
+            run(cmdline=["sqop", "generate-key", "--signing-only", "Test"], stdout=o)
+
+        signing_key.chmod(0o755)
+
+        # extract public key (certificate)
+        with open(signing_key, "rb") as i, open(signing_cert, "wb") as o:
+            run(cmdline=["sqop", "extract-cert"], stdin=i, stdout=o)
+
+        signing_cert.chmod(0o755)
+
+        image.build(
+            options=["--checksum=true", "--openpgp-tool=sqop", "--sign=true", f"--key={signing_key}"]
+        )
+
+        signed_file = image.output_dir / "image.SHA256SUMS"
+        signature = image.output_dir / "image.SHA256SUMS.gpg"
+
+        with open(signed_file, "rb") as i:
+            run(cmdline=["sqop", "verify", signature, signing_cert], stdin=i)
+
+
+def test_signing_checksums_with_gpg(config: ImageConfig) -> None:
+    with tempfile.TemporaryDirectory() as path, Image(config) as image:
+        tmp_path = Path(path)
+        tmp_path.chmod(0o777)
```

Comment:
Let's really not have a world-writable thing, even in tests.

Reply:
The issue here is that `gpg` during its operation writes temporary files to that directory and since the tests starts as root but then switch user during `image.build` the directory must be readable to both root and that temporary build user.

I'll try to minimize the number of `chmod`s. With the `sop` case the key and cert files need to be just readable by both users but in GPG case it needs writeable access sadly :/